common/js/maps.ts

```diff
@@ -0,0 +1,37 @@
+// Utility functions for use on Maps
```

Comment:
Not a fan of the "js" folder. First because these are TS, not JS, and secondly because this is an huge catch-all: every piece of code we write that isn't Rust could seemingly match. These are all dealing with data structure implementations, so `data-structures` might make sense if you really don't want them to be top-level in `common`